README.md

```diff
@@ -48,95 +53,117 @@ defmodule MyProject.Mixfile do
-interface module that performs some conversions and makes calling remote
-functions easier. You will not directly interface with the static module,
-but it is the one that's started when `start_link` is called.
-
-The static client module uses [James Fish](http://github.com/fishcakez)'s
-excellent [connection](http://github.com/fishcakez/connection) behaviour.
+## Clients
 
-For each function defined in the service, the generated module has
-four functions.
+You interact with Thrift services using generated, service-specific interface
+modules. These modules handle type conversions and make calling the service's
+remote functions easier.
 
-Function name  | Description
----------------|----------
-`get_user_by_id/3` | Makes a request to the remote `get_user_by_id` RPC. You can optionally pass `gen_tcp` and `GenServer` options (such as a timeout) to the client as the final `rpc_opts` argument. Returns `{:ok, response}` or `{:error, reason}` tuples.
-`get_user_by_id!/3`  | Same as above, but raises an exception if something goes wrong. The type of exception can be one of the exceptions defined in the service or `Thrift.TApplicationException`.
+```elixir
+iex> alias Calculator.Generated.Service.Binary.Framed.Client
+iex> {:ok, client} = Client.start_link("localhost", 9090, [])
+iex> {:ok, result} = Client.add(client, 10, 20)
+{:ok, 30}
+```
 
-To use the client, simply call `start_link`, supplying the host and port.
+We generate two versions of each function defined by the Thrift service's
+interface: one that returns a standard result tuple, and a `!` variant that
+returns a single result value but raises an exception if an error occurs.
 
 ```elixir
-iex> alias Thrift.Test.UserService.Binary.Framed.Client
-iex> {:ok, client} = Client.start_link("localhost", 2345, [])
-iex> {:ok, user} = Client.get_user_by_id(client, 22451)
-{:ok, %Thrift.Test.User{id: 22451, username: "stinky", first_name: "Stinky", last_name: "Stinkman"}}
+@spec add(pid(), integer(), integer(), Client.options()) :: {:ok, integer()} | {:error, any()}
+def add(client, left, right, rpc_opts \\ [])
+
+@spec add!(pid(), integer(), integer(), Client.options()) :: integer()
+def add!(client, left, right, rpc_opts \\ [])
 ```
 
-## Using The Server
-Creating a thrift server is slightly more involved than creating the client, because
-you need to create a module to handle the work. Fortunately, Elixir Thrift
-creates a [Behaviour](http://elixir-lang.org/docs/stable/elixir/behaviours.html#content),
-complete with correct success typing, for this module. To implement this behaviour,
-use the `@behaviour` module attribute. The compiler will now inform you about
-any missed functions.
+## Servers
 
-Here is an implementation for the server defined above:
+In order to start a Thrift server, you will need to provide a callback module
+that implements the functions described by its service interface. Fortunately,
+a [behaviour][] module will be automatically generated for you, complete with
```

Comment:
I don't think you need the extra `[]` in your link here, at least according to [this markdown cheatsheet](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links).